app_allocator/classes/assignments.py

```diff
@@ -1,9 +1,59 @@
-assignments = {}
+def assign(judge, application):
+    if application:
+        judge.add_application(application)
+        application.assign_judge(judge)
 
 
-def assign(judge, startup):
-    if startup:
-        judge.add_startup(startup)
-        judge_assignments = assignments.get(judge.id(), set())
-        judge_assignments.add(startup.id())
-        assignments[judge.id()] = judge_assignments
+def can_be_assigned(judge, application):
+    if application in judge.all_applications:
+        return False
+    count = application.zscore_count
```

Comment:
From logical standpoint, if there are no values to base zscore on, does it make sense to compute the expected_zscore?  The logic for this code clearly states if we don't have any judging data yet, then we take any judge regardless of their zscore.

Also conceptually, expected_zscore and estimated_zscore are the heaviest part of this so if we can avoid computing them that seems like a win.  In the current case they are trivial, but those could become more complicated, even involved database queries.

Reply:
So here's what I'm thinking - modified it a little to avoid the extra computations
This has not been tested, so modify as needed to account for any blunders. 

```
def can_be_assigned(judge, application):
    if application in judge.all_applications:
        return False
    return zscore_check(application, judge)

def zscore_check(application, judge):
    count = application.zscore_count
    if count == 0:
        return True
    expected = application.expected_zscore(judge.zscore())
    current = application.estimated_zscore()
    count_fn = COUNT_ZSCORE_LOOKUP.get(count, additional_reads)
    return count_fn(current, expected)
```

The main thing for me is that as written, `can_be_assigned` feels cluttered up logic that feels like it's just supporting `zscore_check`.

Reply:
Seems like a matter of personal taste.  To me the above feels like zscore_check isn't really checking a Z-score and all can_be_assigned is doing is a single guard check.